src/xdp-utils.c

```diff
@@ -524,24 +687,20 @@ xdp_app_info_get_path_for_fd (XdpAppInfo *app_info,
 }
 
 static gboolean
-is_valid_initial_name_character (gint c, gboolean allow_dash)
+is_valid_name_character (gint c, gboolean allow_dash)
 {
   return
     (c >= 'A' && c <= 'Z') ||
     (c >= 'a' && c <= 'z') ||
+    (c >= '0' && c <= '9') ||
     (c == '_') || (allow_dash && c == '-');
 }
 
-static gboolean
-is_valid_name_character (gint c, gboolean allow_dash)
-{
-  return
-    is_valid_initial_name_character (c, allow_dash) ||
-    (c >= '0' && c <= '9');
-}
-
+/* This is the same as flatpak apps, except we also allow
+   names to start with digits, so that ids of the form
+   snap.pkg.$snapname is allowed for all snap names. */
```

Comment:
I used "snap.pkg.*" as the prefix to work around the fact that xdp_is_valid_app_id required at least three dotted components.

I can keep using that, but if we're modifying the validation rules what do you think of allowing two dotted components and shortening it to "snap.$snapname"?

Reply:
I'm fine with allowing two components.